index.js

```diff
@@ -627,6 +643,8 @@ const compile = function(schema, root, reporter, opts, scope) {
 
     if (node.maxItems !== undefined) {
       if (!Number.isFinite(node.maxItems)) throw new Error('Invalid maxItems')
+      if (Array.isArray(node.items) && node.items.length > node.maxItems)
```

Comment:
```suggestion
      if (Array.isArray(node.items) && node.items.length !== node.maxItems)
```
It could also be the other way around, we'd want to throw on that right?

Reply:
In general, shouldn't maxItems be pointless if `node.items` is an array?

Reply:
No. This is a logical error in the schema which likely makes it broken.

`maxItems` < `items.length` is not a logical item -- but only first `items.length` items would be validated.
https://json-schema.org/understanding-json-schema/reference/array.html#id6
> And, by default, it’s also okay to add additional items to end:
> The `additionalItems` keyword controls whether it’s valid to have additional items in the array beyond what is defined in items. 

That is separately checked in `strong` mode only.

Reply:
https://github.com/ExodusMovement/is-my-json-valid/pull/28/files#diff-168726dbe96b3ce427e7fedce31bb0bcR300